.gitignore

```diff
@@ -13,3 +13,6 @@ packages
 # Include when developing application packages.
 pubspec.lock
 .packages
+
+# Don't commit IntelliJ project directories
+.idea
```

Comment:
Ignore things that are part of the core tool set (like files created by pub).

Not files specific to your tool set (like your IDE)